main/src/ca/uwaterloo/flix/language/phase/Inliner.scala

```diff
@@ -305,4 +317,25 @@ object Inliner {
 
     case Expression.MatchError(_, _) => exp0
   }
+
+  /**
+   * returns `true` if `exp0` is considered a trivial expression
+   */
+  private def isTrivialExp(exp0: Expression): Boolean = exp0 match {
+    case Expression.Unit(_) => true
+    case Expression.Null(_, _) => true
+    case Expression.True(_) => true
+    case Expression.False(_) => true
+    case Expression.Char(_, _) => true
+    case Expression.Float32(_, _) => true
+    case Expression.Float64(_, _) => true
+    case Expression.Int8(_, _) => true
+    case Expression.Int16(_, _) => true
+    case Expression.Int32(_, _) => true
+    case Expression.Int64(_, _) => true
+    case Expression.BigInt(_, _) => true
+    case Expression.Str(_, _) => true
+    case Expression.Var(_, _, _) => true
```

Comment:
These should occur in the same order as in the AST, e.g. the Var case should be before unit.

Reply:
What AST are you looking in? It should be the same order as both Lifted and Occurence Ast

Reply:
You are right. That is strange. OK leave it as is then!